src/main/scala/cromwell/engine/backend/BackendCall.scala

```diff
@@ -54,21 +54,37 @@ trait ExecutionHandle {
   def result: ExecutionResult
 }
 
+sealed trait ReturnCode {
+  def code: Int
+
+  def asScriptReturnCode = this match {
+    case script: ScriptReturnCode => Option(code)
+    case _ => None
+  }
+
+  def asBackendReturnCode = this match {
+    case script: BackendReturnCode => Option(code)
+    case _ => None
+  }
```

Comment:
I can get rid of the `asXXXReturnCode` which I agree are a bit weird because it requires the sealed trait to have knowledge of its implementations which I don't like either now that I think about it.
However I'm not 100% clear on whether we want to/can have both a Backend code and a Script code